### PR TITLE
feat: add Keep3rGasSaver token

### PR DESCRIPTION
**Pull request by steffenix**

### CODE REVIEW

File: contracts/sugar-mommy/Keep3rGasSaver.sol [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+interface IFreeFromUpTo {
+    function freeFromUpTo(address from, uint256 value)
+        external
+        returns (uint256 freed);
+}
+abstract contract Keep3rGasSaver {
+    using SafeMath for uint256;
+    IFreeFromUpTo chi = IFreeFromUpTo(0x0000000000004946c0e9F43F4Dee607b0eF1fA1c);
+    IFreeFromUpTo gst2 = IFreeFromUpTo(0x0000000000b3F879cb30FE243b4Dfee438691c04);
```
Comment (lbertenasco) [inline contracts/sugar-mommy/Keep3rGasSaver.sol:11]:
```suggestion
    address public constant chi = address(0x0000000000004946c0e9F43F4Dee607b0eF1fA1c);
    address public constant gst2 = address(0x0000000000b3F879cb30FE243b4Dfee438691c04);
```

**Comment by steffenix** [comment]
Should probably add https://github.com/matnad/liquid-gas-token/blob/master/contracts/LgtHelper.sol

**Comment by lbertenasco** [comment]
If we could have a token agnostic contract that we could use with any gas token it'd be awesome :)

**Comment by steffenix** [comment]
LGT added.

**Comment by lbertenasco** [comment]
Nice one @steffenix ! MVP looks good.

Now we need a way to add new gas tokens without having to redeploy our contracts.

One idea is to do this by using:
`abstract Keep3rGasSaver` on our contracts: but have it connect directly into a :
- `GasTokensRegistry.saveOnGas(_gas_token)` which in turn will have a mapping of:
- `gas_token => gas_token_wrapper` where the token wrapper is just a stateless contract that has `start` and `end` functions.

Do gas tokens need to be approved to the contract that's going to use them?
if so, we'll need to think what's the safest way to do this

what do you think?

**Comment by steffenix** [comment]
I don't think they will be that many new gas tokens added along the way.
They are good to use until eip-1559

https://hackmd.io/@HWeNw8hNRimMm2m2GH56Cw/evm_feature_removing#Gas-refunds

They need to be approved yes.

**Comment by steffenix** [comment]
https://github.com/ethereum/EIPs/pull/3298

**Pull request closed without merging**